pkg/controllers/user/logging/clusterlogging.go

```diff
@@ -0,0 +1,102 @@
+package logging
+
+import (
+	"github.com/pkg/errors"
+	"github.com/rancher/types/apis/apps/v1beta2"
+	"github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	rbacv1 "github.com/rancher/types/apis/rbac.authorization.k8s.io/v1"
+	"github.com/rancher/types/config"
+	"k8s.io/apimachinery/pkg/labels"
+
+	loggingconfig "github.com/rancher/rancher/pkg/controllers/user/logging/config"
+	"github.com/rancher/rancher/pkg/controllers/user/logging/generator"
+	"github.com/rancher/rancher/pkg/controllers/user/logging/utils"
+)
+
+type ClusterLoggingSyncer struct {
+	clusterLoggings      v3.ClusterLoggingInterface
+	projectLoggingLister v3.ProjectLoggingLister
+	services             v1.ServiceInterface
+	serviceAccounts      v1.ServiceAccountInterface
+	configmaps           v1.ConfigMapInterface
+	daemonsets           v1beta2.DaemonSetInterface
+	deployments          v1beta2.DeploymentInterface
+	roles                rbacv1.RoleInterface
+	rolebindings         rbacv1.RoleBindingInterface
+	clusterRoleBindings  rbacv1.ClusterRoleBindingInterface
+}
+
+func registerClusterLogging(cluster *config.UserContext) {
+	clusterloggingClient := cluster.Management.Management.ClusterLoggings("")
+	syncer := &ClusterLoggingSyncer{
+		clusterLoggings:      clusterloggingClient,
+		projectLoggingLister: cluster.Management.Management.ProjectLoggings("").Controller().Lister(),
+		services:             cluster.Core.Services(loggingconfig.LoggingNamespace),
+		serviceAccounts:      cluster.Core.ServiceAccounts(loggingconfig.LoggingNamespace),
+		configmaps:           cluster.Core.ConfigMaps(loggingconfig.LoggingNamespace),
+		daemonsets:           cluster.Apps.DaemonSets(loggingconfig.LoggingNamespace),
+		deployments:          cluster.Apps.Deployments(loggingconfig.LoggingNamespace),
+		roles:                cluster.RBAC.Roles(loggingconfig.LoggingNamespace),
+		rolebindings:         cluster.RBAC.RoleBindings(loggingconfig.LoggingNamespace),
+		clusterRoleBindings:  cluster.RBAC.ClusterRoleBindings(loggingconfig.LoggingNamespace),
+	}
+	clusterloggingClient.AddClusterScopedHandler("cluster-logging-controller", cluster.ClusterName, syncer.Sync)
+}
+
+func (c *ClusterLoggingSyncer) Sync(key string, obj *v3.ClusterLogging) error {
+	//clean up
+	if obj == nil || obj.DeletionTimestamp != nil {
+		if err := utils.RemoveEmbeddedTarget(c.deployments, c.serviceAccounts, c.services, c.roles, c.rolebindings); err != nil {
+			return err
+		}
+
+		allDisabled, err := utils.IsAllLoggingDisable(c.clusterLoggings.Controller().Lister(), c.projectLoggingLister)
+		if err != nil {
+			return err
+		}
+
+		if allDisabled {
+			if err := utils.RemoveFluentd(c.daemonsets, c.serviceAccounts, c.clusterRoleBindings); err != nil {
+				return err
+			}
+		}
```

Comment:
looks like we return when obj == nil || obj.DeletionTimestamp != nil  only when error happens. We should return regardless